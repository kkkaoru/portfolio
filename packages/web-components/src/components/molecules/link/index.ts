export * from './list';
export * from './contact';
