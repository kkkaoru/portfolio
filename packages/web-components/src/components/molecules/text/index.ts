export * from './title-and-description';
