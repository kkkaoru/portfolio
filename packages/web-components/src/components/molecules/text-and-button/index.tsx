/* eslint-disable @typescript-eslint/ban-ts-comment */
// Cannot safe by tsc when using in web-client
// @ts-ignore
import { DefaultButton, Props as ButtonProps } from '@web-components/components/atoms/button';
// @ts-ignore
import { TextWithUnderline } from '@web-components/components/atoms/text';

export type Props = {
  textTitle: string;
  buttonLabel: string;
  buttonProps?: Omit<ButtonProps, 'label'>;
};

export function TextAndButton({ textTitle, buttonLabel, buttonProps }: Props): JSX.Element {
  return (
    <div>
      <TextWithUnderline text={textTitle} />
      <DefaultButton label={buttonLabel} {...buttonProps} />
    </div>
  );
}
