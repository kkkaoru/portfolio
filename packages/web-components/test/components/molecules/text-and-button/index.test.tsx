import { render } from '@testing-library/react';
import { TextAndButton } from '@web-components/components/molecules/text-and-button';

describe('example', () => {
  it('should be matched snapshot', () => {
    const { asFragment } = render(<TextAndButton textTitle="snap shot title" buttonLabel="snap shot label" />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
