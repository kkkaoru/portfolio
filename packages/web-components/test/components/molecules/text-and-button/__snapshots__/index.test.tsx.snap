// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example should be matched snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="flex flex-col items-center justify-center w-full p-10 font-bold break-all text-7xl lg:text-9xl after:block after:w-full after:h-1 after:bg-red-100"
    >
      snap shot title
    </div>
    <button
      class="font-sans font-bold border-0 rounded-[3em] cursor-pointer inline-block leading-none text-base py-[11px] px-[20px] text-[#333] bg-transparent shadow-md"
      type="button"
    >
      snap shot label
    </button>
  </div>
</DocumentFragment>
`;
